### ISSUE TITLE
PyTorch macosx-arm64 cross-build

### ISSUE THREAD

**Issue by sbrunk** [bytedeco/javacpp-presets#1461]
TL;DR
Can we use the x86_64 Python interpreter for cross-compilation of PyTorch to macosx-arm64 or will it cause issues? If we can, what is a good way to configure it especially for CI builds? If not, what might be other options?

## Background

We only have x86_64 machines in CI, so we currently have to cross-compile macosx-arm64 artifacts. With https://github.com/bytedeco/javacpp-presets/issues/1459 out of the way, I got cross-compilation PyTorch working locally for #1069. I can compile arm64 artifacts on x86_64 that are working fine on an arm64 machine. 

What I'm not sure about though is the Python build dependency. One thing it's used for is to call the build:

https://github.com/bytedeco/javacpp-presets/blob/ee1401da0b612ad515f06fa6846937ebb87f2a18/pytorch/cppbuild.sh#L185

Which means for cross-compilation we need an interpreter of the **host** arch (x86_64 here). It is also used for other things though, i.e. and for numpy stuff and in `CFLAGS` here:

https://github.com/bytedeco/javacpp-presets/blob/ee1401da0b612ad515f06fa6846937ebb87f2a18/pytorch/cppbuild.sh#L105-L105

The PyTorch build depends on cpython of its **target** arch though. Locally via 

https://github.com/bytedeco/javacpp-presets/blob/ee1401da0b612ad515f06fa6846937ebb87f2a18/pytorch/cppbuild.sh#L59-L59

For a maven build it seems `BUILD_PATH` is set and therefore `CPYTHON_PATH` is overwritten to use cpython cached from maven:

https://github.com/bytedeco/javacpp-presets/blob/ee1401da0b612ad515f06fa6846937ebb87f2a18/pytorch/cppbuild.sh#L63-L68

I'm not sure how `CPYTHON_PATH` is set exactly (maven plugin?) but I think it is coming in through the  transitive  `maven-compiler-plugin` dependency on cpython via `numpy-platform` in `pom.xml`:

https://github.com/bytedeco/javacpp-presets/blob/ee1401da0b612ad515f06fa6846937ebb87f2a18/pytorch/pom.xml#L52-L56

To get it working locally, I did a little hack: Build a native x86_64 cpython but put it into the `cppbuild/macosx-arm64` directory so `CPYTHON_PATH` is set to it and we can actually run that Python interpreter for kicking off the PyTorch cross-compilation.

So the open question is: Can we use the x86_64 Python interpreter for cross-compilation of PyTorch or will it cause issues? If we can, what is a good way to configure it especially for CI builds? If not, what might be other options?
